src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,178 @@
+package frc.trigon.robot.subsystems.arm;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.*;
+import frc.trigon.robot.utilities.Conversions;
+
+public class Arm extends SubsystemBase {
+    private final static Arm INSTANCE = new Arm();
+    private final CANSparkMax
+            masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+
+    }
+
+
+    public Command setTargetState(ArmConstants.ArmState targetState) {
+        if (targetState.elevatorPosition >= getElevatorPositionRevolutions()) {
+            return new SequentialCommandGroup(
+                    getSetTargetAngleCommand(targetState.angle),
+                    getSetTargetElevatorPositionCommand(targetState.elevatorPosition)
+            );
+        } if (targetState.elevatorPosition < getElevatorPositionRevolutions()) {
+            return new SequentialCommandGroup(
+                    getSetTargetElevatorPositionCommand(targetState.elevatorPosition),
+                    getSetTargetAngleCommand(targetState.angle)
+            );
+        }
+    }
```

Comment:
There is a slight problem here. The command getter will be called at the start of the code, not when the command should be run. This means `targetState.elevatorPosition >= getElevatorPositionRevolutions()` won't be dynamic to the current position, just a statement decided at the start of the code. To solve this, use a DefferedCommand. Read a bit about it.

Reply:
I couldn't find a great source to find out what this is. Do I have to add @Deprecated before the command?

Reply:
Sorry for my small typo, I meant DeferredCommand. Type it in intellij and read its javadocs (you can press ctrl + enter to enter the class)
@depercated is a whole other unrelated thing.

Reply:
When I put it in it says it "can't resolve symbol"